In [2]:
import joblib

model = joblib.load(r'C:/Users/sarav/OneDrive/Área de Trabalho/Trabalhos/LightHouse/LH_CD_SARA/models/imdb_model.pkl')


In [3]:
import pandas as pd

cidade_de_deus = {
    'Released_Year': 2002,
    'Runtime': 130,
    'Meta_score': 79.0,
    'Gross': 7563397,
    'No_of_Votes': 699256,
    'Certificate': 'A',
    'Genre': 'Crime, Drama',
    'Overview_clean': 'in the slums of rio two kids paths diverge as one struggles to become a photographer and the other a kingpin'
}

df_exemplo = pd.DataFrame([cidade_de_deus])


In [5]:
df_exemplo_renomeado = df_exemplo.rename(columns={'Overview_clean': 'Cleaned_Overview'})
nota_prevista = model.predict(df_exemplo_renomeado)[0]
print(f"Nota prevista pelo modelo: {nota_prevista:.2f}")


Nota prevista pelo modelo: 8.30


In [6]:
erro_medio = 0.245  

intervalo = (nota_prevista - erro_medio, nota_prevista + erro_medio)
print(f"Intervalo de confiança: {intervalo[0]:.2f} a {intervalo[1]:.2f}")


Intervalo de confiança: 8.06 a 8.55


In [7]:
nota_real = 8.6
diferenca = abs(nota_prevista - nota_real)

print(f"Nota real: {nota_real}")
print(f"Erro absoluto: {diferenca:.2f}")


Nota real: 8.6
Erro absoluto: 0.30


## Justificativa da Previsão

O modelo previu uma nota de **8.30** para o filme “Cidade de Deus”, com intervalo de confiança entre **8.06** e **8.55**. A nota real do IMDB é **8.6**, resultando em um erro absoluto de **0.30**.

Essa previsão foi baseada em variáveis como:
- Gênero: Crime, Drama
- Duração: 130 minutos
- Meta Score: 79
- Número de votos: 699 mil
- Sinopse: Indicativa de temas sociais, violência urbana e trajetória de personagens

O modelo demonstrou boa capacidade de generalização ao se aproximar da nota real, reforçando sua utilidade para prever avaliações futuras.


# 🎬 Extra — Predição do Filme “O Morro dos Ventos Uivantes” (2026)

Com base nas informações divulgadas até agora: 

In [8]:
morro_dos_ventos_uivantes = {
    'Released_Year': 2026,
    'Runtime': 125,
    'Meta_score': 82.0,
    'Gross': 85000000,  # estimativa para blockbuster internacional
    'No_of_Votes': 500000,  # estimativa inicial
    'Certificate': 'A',
    'Genre': 'Romance, Drama',
    'Overview_clean': (
        'O tragico e obsessivo romance entre catherine e heathcliff ambientado na inglaterra do século dezenove '
        'explora temas de desejo, vingança e turbulência emocional em uma adaptação moderna e sensual'
    )
}

df_extra = pd.DataFrame([morro_dos_ventos_uivantes])


In [10]:
df_extra_renomeado = df_extra.rename(columns={'Overview_clean': 'Cleaned_Overview'})
nota_prevista_extra = model.predict(df_extra_renomeado)[0]
print(f"Nota prevista para 'O Morro dos Ventos Uivantes': {nota_prevista_extra:.2f}")


Nota prevista para 'O Morro dos Ventos Uivantes': 7.84


In [11]:
erro_medio = 0.245
intervalo_extra = (
    nota_prevista_extra - erro_medio,
    nota_prevista_extra + erro_medio
)
print(f"Intervalo de confiança: {intervalo_extra[0]:.2f} a {intervalo_extra[1]:.2f}")


Intervalo de confiança: 7.60 a 8.09


## Justificativa Técnica da Predição 

O modelo previu uma nota de **7.84** para o filme “O Morro dos Ventos Uivantes”, com intervalo de confiança entre **7.60** e **8.09**. A previsão foi feita com base em variáveis como:

- Gênero: Romance, Drama
- Duração: 125 minutos
- Meta Score estimado: 82
- Número de votos estimado: 500 mil
- Sinopse: Adaptação moderna e intensa de um clássico literário
- Elenco: Margot Robbie e Jacob Elordi
- Produção: Warner Bros.

Essas características indicam forte potencial de avaliação positiva. A previsão reforça a utilidade do modelo para orientar decisões estratégicas sobre futuros lançamentos.


## Reflexão Crítica: Representatividade e Apagamento Histórico

Apesar da previsão técnica indicar uma boa recepção, é importante considerar fatores que o modelo não capta. Heathcliff, personagem central da obra original, é descrito com pele escura e traços associados à imagem estereotipada de um cigano, e isso é essencial para entender o preconceito que ele sofre.

Na adaptação, essa dimensão é apagada ao escalar um ator branco e “neutro”, promovendo uma estética que embranquece narrativas e distorce o conflito racial presente no texto original. Essa escolha não é apenas estética, é política. Ela contribui para um apagamento histórico que afeta a forma como novas gerações interpretam a obra.

Esse tipo de decisão pode impactar negativamente a recepção crítica e pública do filme, especialmente entre espectadores que valorizam representatividade e fidelidade cultural. Portanto, embora o modelo preveja uma nota positiva, a realidade pode ser mais complexa.
